crates/router/src/core/mandate/utils.rs

```diff
@@ -0,0 +1,78 @@
+use std::marker::PhantomData;
+
+use common_utils::{errors::CustomResult, ext_traits::ValueExt};
+use diesel_models::Mandate;
+use error_stack::ResultExt;
+
+use crate::{
+    core::{errors, payments::helpers},
+    types::{self, api::mandates, domain, PaymentAddress},
+};
+const IRRELEVANT_PAYMENT_ID_IN_MANDATE_REVOKE_FLOW: &str =
+    "irrelevant_payment_id_in_mandate_revoke_flow";
+
+const IRRELEVANT_ATTEMPT_ID_IN_MANDATE_REVOKE_FLOW: &str =
+    "irrelevant_attempt_id_in_mandate_revoke_flow";
+
+const IRRELEVANT_CONNECTOR_REQUEST_REFERENCE_ID_IN_MANDATE_REVOKE_FLOW: &str =
+    "irrelevant_connector_request_reference_id_in_mandate_revoke_flow";
+
+pub async fn construct_mandate_revoke_router_data(
+    connector_name: &str,
+    request: mandates::MandateId,
+    merchant_connector_account: helpers::MerchantConnectorAccountType,
+    merchant_account: &domain::MerchantAccount,
+    mandate: Mandate,
+) -> CustomResult<types::MandateRevokeRouterData, errors::ApiErrorResponse> {
+    let auth_type: types::ConnectorAuthType = merchant_connector_account
+        .get_connector_account_details()
+        .parse_value("ConnectorAuthType")
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+    let router_data = types::RouterData {
+        flow: PhantomData,
+        merchant_id: merchant_account.merchant_id.clone(),
+        customer_id: Some(mandate.customer_id),
+        connector_customer: None,
+        connector: connector_name.to_string(),
+        payment_id: mandate
+            .original_payment_id
+            .unwrap_or_else(|| IRRELEVANT_PAYMENT_ID_IN_MANDATE_REVOKE_FLOW.to_string()),
```

Comment:
Do we want to raise an error here if original_payment_id is not found @vspecky?

Reply:
There is a check in `revoke_mandate` function for this, if that check is not passed the pointer will not reach this point, so this is majorly a fallback to comply with the types and can be ignored